src/adapter/src/catalog.rs

```diff
@@ -2738,12 +2738,24 @@ impl Catalog {
             (system_config, boot_ts)
         };
         for (name, value) in &bootstrap_system_parameters {
-            if !system_config.contains_key(name) {
-                self.state.insert_system_configuration(name, value)?;
-            }
+            let result = self.state.insert_system_configuration(name, value);
+            result.or_else(|e| match e {
+                AdapterError::UnknownParameter(name) => {
+                    tracing::warn!("Cannot load unknown system parameter `{}` from stash", name);
+                    Ok(false)
+                }
+                err => Err(err),
+            })?;
         }
         for (name, value) in system_config {
-            self.state.insert_system_configuration(&name, &value)?;
+            let result = self.state.insert_system_configuration(&name, &value);
+            result.or_else(|e| match e {
+                AdapterError::UnknownParameter(name) => {
+                    tracing::warn!("Cannot load unknown system parameter `{}` from stash", name);
+                    Ok(false)
+                }
+                err => Err(err),
+            })?;
```

Comment:
Weak preference for using just a single `match` statement. I at least found the match inside the `or_else` to be kind of a head scratcher. Might jsut be me.

```patch
diff --git a/src/adapter/src/catalog.rs b/src/adapter/src/catalog.rs
index b8ee98eeab..bef84dc00b 100644
--- a/src/adapter/src/catalog.rs
+++ b/src/adapter/src/catalog.rs
@@ -27,7 +27,7 @@ use once_cell::sync::Lazy;
 use regex::Regex;
 use serde::{Deserialize, Serialize};
 use tokio::sync::{mpsc, Mutex, MutexGuard};
-use tracing::{info, trace};
+use tracing::{info, warn, trace};
 use uuid::Uuid;
 
 use mz_audit_log::{
@@ -2738,24 +2738,22 @@ impl Catalog {
             (system_config, boot_ts)
         };
         for (name, value) in &bootstrap_system_parameters {
-            let result = self.state.insert_system_configuration(name, value);
-            result.or_else(|e| match e {
-                AdapterError::UnknownParameter(name) => {
-                    tracing::warn!("Cannot load unknown system parameter `{}` from stash", name);
-                    Ok(false)
+            match self.state.insert_system_configuration(name, value) {
+                Ok(_) => (),
+                Err(AdapterError::UnknownParameter(name)) => {
+                    warn!(%name, "cannot load unknown system parameter from stash");
                 }
-                err => Err(err),
-            })?;
+                Err(e) => return Err(e),
+            }
         }
         for (name, value) in system_config {
-            let result = self.state.insert_system_configuration(&name, &value);
-            result.or_else(|e| match e {
-                AdapterError::UnknownParameter(name) => {
-                    tracing::warn!("Cannot load unknown system parameter `{}` from stash", name);
-                    Ok(false)
+            match self.state.insert_system_configuration(&name, &value) {
+                Ok(_) => (),
+                Err(AdapterError::UnknownParameter(name)) => {
+                    warn!(%name, "cannot load unknown system parameter from stash");
                 }
-                err => Err(err),
-            })?;
+                Err(e) => return Err(e),
+            }
         }
         if let Some(system_parameter_frontend) = system_parameter_frontend {
             if !self.state.system_config().config_has_synced_once() {
```

One other style nits in the patch: using a structured argument for the `name` parameter. We're moving in that direction generally (see STYLE.md). Will make it easy to search the logs for specific instances of unknown system parameters, which could in theory be useful.

Reply:
I had the suggested code in my first version, but thought the `or_else` is simpler to read because it has only two arms and no explicit `return Err`. There are a lot of other `.or_else` calls in the code, but I guess most are on the `Option` variant.

Reply:
Accepted the change.